specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-04-01-preview/managedHsm.json

```diff
@@ -0,0 +1,634 @@
+              },
+              {
+                "value": "SecurityDomainRestore",
+                "description": "The managed HSM pool is waiting for a security domain restore action."
+              },
+              {
+                "value": "Restoring",
+                "description": "The managed HSM pool is being restored from full HSM backup."
+              },
+              {
+                "value": "Updating",
+                "description": "The managed HSM Pool is currently being updated."
+              }
+            ]
+          }
+        }
+      },
+      "description": "Properties of the managed HSM Pool"
+    },
+    "ManagedHsm": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ManagedHsmProperties",
+          "description": "Properties of the managed HSM"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ManagedHsmResource"
+        }
+      ],
+      "description": "Resource information with extended details."
+    },
+    "ManagedHsmResource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The Azure Resource Manager resource ID for the managed HSM Pool."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the managed HSM Pool."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type of the managed HSM Pool."
+        },
+        "location": {
+          "type": "string",
+          "description": "The supported Azure location where the managed HSM Pool should be created.",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "sku": {
+          "$ref": "#/definitions/ManagedHsmSku",
+          "description": "SKU details"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "Managed HSM resource",
+      "x-ms-azure-resource": true
+    },
+    "ManagedHsmListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedHsm"
+          },
+          "description": "The list of managed HSM Pools."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of managed HSM Pools."
+        }
+      },
+      "description": "List of managed HSM Pools"
+    },
+    "ManagedHsmError": {
+      "properties": {
+        "error": {
+          "readOnly": true,
+          "description": "The server error.",
+          "$ref": "#/definitions/Error"
+        }
+      },
+      "description": "The error exception."
+    },
+    "Error": {
```

Comment:
Why do we need the above ManagedHsmError? Why don't we just rename this Error to ManagedHsmError? Are you following some suggested pattern here?

Reply:
Yeah, we are having our own name in case we want to add any more properties to it in future.